app/routes/application.js

```diff
@@ -32,5 +35,16 @@ export default Ember.Route.extend(OSFAgnosticAuthRouteMixin, {
         if (locale) {
             this.set('i18n.locale', locale);
         }
+
+        // Check permissions
```

Comment:
This seems like it'd make more sense in `beforeModel`, so it won't try fetching all the models before redirecting to `index`.

Reply:
I had it in `beforeModel` originally but session wasn't defined yet so it always said the user wasn't logged in.

Reply:
Aw, that's annoying